ext/pathname/lib/pathname.rb

```diff
@@ -503,7 +503,7 @@ def each_child(with_directory=true, &b)
   # ArgumentError is raised when it cannot find a relative path.
   #
   def relative_path_from(base_directory)
-    base_directory = Pathname.new(base_directory) unless Pathname === base_directory
+    base_directory = Pathname.new(base_directory) unless base_directory.is_a?(Pathname)
```

Comment:
Whether this uses `===` or `is_a?` is not essential. What's the problem here is dde0e30c2dc98d8356374a1173c7129fbc8e5e14 changed the type of `base_directory`. Previous type is "an object which respond_to `clean_path`" and changed to "an object which is kind of `Pathname` or an object which can be an argument of `Path.new`".
dde0e30c2dc98d8356374a1173c7129fbc8e5e14 must change to "an object which respond_to `clean_path` or an object which can be an argument of `Path.new`".

Reply:
> Whether this uses === or is_a? is not essential.

Actually, it does fix the bug, because `is_a?` is correctly handled by the mock (i.e. `mock.is_a?(Pathname)` is true even if it is lying).

But you are right, the requirements on the argument changed.

There is more than one way to solve, this, but if we can solve it for RC2, that would be great. Do you think you can implement your preferred solution?

Reply:
> Actually, it does fix the bug, because is_a? is correctly handled by the mock (i.e. mock.is_a?(Pathname) is true even if it is lying).

Where is the actual code of the library?

Anyway RC2 is already begun its process. It's too late.

Reply:
FYI, "its process" means tag the commit, make tarball, commit release article to web repo, and so on.

Reply:
Thanks for explaining.